actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,335 @@
+package org.sunbird.learner.actors.coursebatch;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang.StringUtils;
+import org.apache.velocity.VelocityContext;
+import org.sunbird.actor.background.BackgroundOperations;
+import org.sunbird.actor.core.BaseActor;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.ActorOperations;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.datasecurity.DecryptionService;
+import org.sunbird.common.models.util.mail.SendMail;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.actors.notificationservice.dao.EmailTemplateDao;
+import org.sunbird.learner.actors.notificationservice.dao.impl.EmailTemplateDaoImpl;
+import org.sunbird.learner.util.Util;
+import org.sunbird.models.course.batch.CourseBatch;
+
+/**
+ * This actor will handle course enrollment operations.
+ *
+ * @author github.com/iostream04
+ */
+
+/** @author mayank */
+@ActorConfig(
+  tasks = {"batchBulk", "batchUpdate", "batchOperation"},
+  asyncTasks = {"batchBulk", "batchUpdate", "batchOperation"}
+)
+public class BatchOperationNotifierActor extends BaseActor {
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+
+  private DecryptionService decryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getDecryptionServiceInstance(
+          null);
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    if (request.getOperation().equalsIgnoreCase(ActorOperations.BATCH_BULK.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
```

Comment:
Map<String, Object> requestMap = request.getRequest(); can be put outside , this line is repeated in all code block except last one.

Reply:
done